File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        
        Arrays.sort(people);
        
        int boat = 0;
        int left = 0;
        int right = people.length - 1;
        
        while(left <= right){
            if(people[left] + people[right] <= limit){
                left++;
            }
            
            right--;
            boat++;
        }
        
        return boat;
        
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    
    people.sort((a, b) => a-b);
    
    var left = 0;
    var right = people.length -1;
    var boat = 0;
    
    while(left <= right){
        
        if(people[left] + people[right] <= limit){
            left ++;
        }
        
        right--;
        boat++;
        
    }
    
    return boat;
}